src/mock_crypto.rs

```diff
@@ -234,6 +236,18 @@ pub(crate) mod crypto_impl {
                 })
             }
 
+            /// Mock seed.
+            #[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Clone)]
+            pub(crate) struct Seed(pub(crate) [u8; SEEDBYTES]);
+
+            /// Generate mock public and corresponding secret key using a seed.
+            pub(crate) fn keypair_from_seed(seed: &Seed) -> (PublicKey, SecretKey) {
+                let pub_key = seed.0;
+                let mut sec_key = [0u8; 64];
+                sec_key[0..32].clone_from_slice(&seed.0); // simply get 64 byte array
+                (PublicKey(pub_key), SecretKey(sec_key))
```

Comment:
Don't we normally make sec-key == pub-key in mock ? pls go over the rest of mock file and confirm.

Also don't see any test cases for this.

Reply:
Well the secret key is 64 bytes while the public key is 32, but the first half of the secret key is equal to the public key, yeah. Added test cases.